### DISCUSSION TITLE
0.2.0-rc1 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#423] [Announcements]
This is the first **release candidate** for version 0.2.0, and assuming no major bugs are found is likely to become 0.2.0-stable.

This release supports [**Godot 4.0.3**](https://github.com/godotengine/godot/releases/tag/4.0.3-stable).

## Known major issues

- Extension classes cause exception when used from C# ([**#270**](https://github.com/godot-jolt/godot-jolt/issues/270))
- The `move_and_collide` parameter `recovery_as_collision` is always `true` ([**#337**](https://github.com/godot-jolt/godot-jolt/issues/337))

## Changelog

These are the notable changes that have been made since [**0.1.0-stable**](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.1.0-stable) was released. You can find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.1.0-stable...v0.2.0-rc1).

Breaking changes are denoted with ⚠️.

### Changed

- ⚠️ Changed friction values to be combined in the same way as in Godot Physics.
- ⚠️ Changed bounce values to be combined in the same way as in Godot Physics.
- ⚠️ Changed the direction of `Generic6DOFJoint` angular motors to match Godot Physics.
- ⚠️ Changed how linear/angular velocities are applied to a frozen `RigidBody3D`, to better match Godot Physics. They now apply a surface velocity, also known as a "constant velocity", instead of actually moving the body.
- ⚠️ Changed shape margins to be interpreted as an upper bound. They are now scaled according to the shape's extents, which removes the ability to have an incorrect margin, thereby removing any warnings about that.
- Changed warning/error messages to provide more context, such as the names of any bodies related/connected to that particular thing.

### Added

- Added new project setting "Bounce Velocity Threshold".
- Added support for the `custom_integrator` property on `RigidBody3D` and `PhysicalBone3D`.
- Added support for the `integrate_forces` method on `PhysicsDirectBodyState3D`.
- Added support for the `rough` and `absorbent` properties on `PhysicsMaterial`.
- Added support for surface velocities, also known as "constant velocities", for both static and kinematic bodies.
- Added support for more flexible limits for `HingeJoint3D` and `SliderJoint3D`.

### Fixed

- Fixed issue where `CharacterBody3D` and other kinematic bodies wouldn't elicit a proper collision response from dynamic bodies.
- Fixed issue where setting friction on an already entered body would instead set bounce.
- Fixed issue where disabling or removing a body connected to a joint would error or crash.
- Fixed issue where `RigidBody3D` would de-sync from its underlying body after freezing with the "Static" freeze mode.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.2.0-rc1'>0.2.0-rc1</a>.</em>

**Comment by mgiraud** [comment]
Congrats on this release !
Just one very small issue i've found so far, the warning messages needs godot to relaunch after switching the physics engine.

**Reply by mihe** [reply]
I've been contemplating submitting a PR to Godot to make the "Physics Engine" setting require a restart of Godot, since it doesn't actually switch the physics engine within the editor until you do, meaning you can technically be running Godot Physics in the editor but be running Godot Jolt in the actual game, which can create some weird situations like the one you mention.

**Reply by mihe** [reply]
I went ahead and submitted a PR for it: godotengine/godot#77839
